plugins/chat/app/views/user_notifications/chat_summary.html.erb

```diff
@@ -2,81 +2,75 @@
-              <%- else %>
-                <%= chat_channel.title(@user) %>
-              <%- end %>
-            </h5>
-          </td>
-        </tr>
-
-        <%- unless SiteSetting.private_email %>
-          <%- messages.take(2).each do |chat_message| %>
-            <%- sender = chat_message.user %>
-            <%- sender_name = @display_usernames ? sender.username : sender.name %>
+  <%- [@grouped_channels, @grouped_dms].each do |grouped_messages| %>
+    <%- grouped_messages.each do |channel, messages| %>
+      <%- other_messages_count = messages.size - 2 %>
+      <table class="chat-summary-content" style="padding:1em;margin-top:20px;width:100%;min-width:100%;background-color:#f7f7f7;">
+        <tbody>
+          <tr>
+            <td colspan="100%">
+              <h5 style="margin:0.5em 0 0.5em 0;font-size:0.9em;">
+                <%- if SiteSetting.private_email %>
+                  <%= I18n.t("system_messages.private_channel_title", id: channel.id) %>
+                <%- else %>
+                  <%= channel.title(@user) %>
+                <%- end %>
+              </h5>
+            </td>
+          </tr>
 
-            <tr class="message-row">
-              <td style="white-space:nowrap;vertical-align:top;padding:<%= rtl? ? '1em 2em 0 0' : '1em 0 0 2em' %>">
-                <img src="<%= sender.small_avatar_url -%>" style="height:20px;width:20px;margin:<%= rtl? ? '0 0 5px 0' : '0 5px 0 0' %>;border-radius:50%;clear:both;display:inline-block;outline:0;text-decoration:none;vertical-align:top;" alt="<%= sender_name -%>">
-                <span style="display:inline-block;color:#0a0a0a;vertical-align:top;font-weight:bold;">
-                  <%= sender_name -%>
-                </span>
-                <span style="display:inline-block;color:#0a0a0a;font-size:0.8em;">
-                  <%= I18n.l(@user_tz.to_local(chat_message.created_at), format: :long) -%>
-                </span>
-              </td>
-            </tr>
-            <tr>
-              <td style="width:99%;margin:0;padding:<%= rtl? ? '0 2em 0 0' : '0 0 0 2em' %>;vertical-align:top;">
-                <%= email_excerpt(chat_message.cooked_for_excerpt) %>
-              </td>
-            </tr>
+          <%- unless SiteSetting.private_email %>
+            <%- messages.take(2).each do |chat_message| %>
+              <%- sender = chat_message.user %>
+              <tr class="message-row">
+                <td style="white-space:nowrap;vertical-align:top;padding:<%= rtl? ? '1em 2em 0 0' : '1em 0 0 2em' %>">
+                  <img src="<%= sender.small_avatar_url -%>" style="height:20px;width:20px;margin:<%= rtl? ? '0 0 5px 0' : '0 5px 0 0' %>;border-radius:50%;clear:both;display:inline-block;outline:0;text-decoration:none;vertical-align:top;" alt="<%= sender.display_name -%>">
+                  <span style="display:inline-block;color:#0a0a0a;vertical-align:top;font-weight:bold;">
+                    <%= sender.display_name -%>
```

Comment:
Use the already defined & well-tested `.display_name` instead of _badly_ duplicating its logic.